RMS time projector

gaurav-ganti opened the pull request:
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/GranthamImperial/silicone/issues/YY>`_)
```

Commit:
First attempt at implementing RMS closest time projector

Commit:
First set of integration tests

Commit:
Updated tests to evaluate accuracy of results

Commit:
Corrected output dataframe formatting issues

Commit:
Added documentation

Commit:
Updated notebooks with extend RMS closest time projector example

Rlamboll reviewed (commented):
Looking good so far on the coding side. Things that need doing still: 
1) Update the changelog to mention this added feature
2) Several lines in the notebook are out-of-date or hard to read. The intro still says there is only one tool, for instance. The legends on the plots could be more useful to explain which of the lines is the infilled one (particularly since it overlaps one of the pre-existing lines) - maybe make the infilled line dashed so we can see both?

Commit:
Changelog updated

Commit:
Fixed deprecation warnings and updated notebook with further description

gaurav-ganti commented:
@Rlamboll 

1. Added
2. Updated the notebook - descriptions fixed, linestyle in plot adjusted to see the underlying pathway, deprecation warnings thrown by pyam addressed by shifting from the ```line_plot()``` to ```plot()```method.

Rlamboll commented on CHANGELOG.rst:
```diff
@@ -5,6 +5,7 @@ Work in progress
 Added
 ~~~~~
 - (`#126 <https://github.com/GranthamImperial/silicone/pull/126>`_) Added the first time projector (Extend latest time quantile) that extends a pathway to cover later times, assuming it remains at the same quantile.
+- (`#129 <https://github.com/GranthamImperial/silicone/pull/129>`_) Added an additional time projector (Extend RMS closest) that extends a pathway to cover later times by selecting future data from the closest pathway.
```
This should go above 126, since it came later. This way people only need to read the top items to get back on track. Description is good though.

gaurav-ganti replied:
Thanks, corrected.

Rlamboll commented on src/silicone/time_projectors/__init__.py:
```diff
@@ -7,3 +7,4 @@
 """
 
 from .extend_latest_time_quantile import ExtendLatestTimeQuantile  # noqa: F401
+from .extend_rms_closest import ExtendRMSClosest
```
Our linter doesn't seem to mind, but adding the # noqa: F401 command means that using a proper linter won't complain that we're not going to get error messages about the fact that these imports aren't used here.

Rlamboll reviewed (approved):
Great, just two minor points and you can merge

Commit:
Corrected order of entry in changelog

Commit:
added #noqa:f401 command
